### PR TITLE
Settings page with Stripe auth and paypal preapproval

### PR DESCRIPTION
**Pull request by asood123**

### CODE REVIEW

File: frontend/src/lib/utils.js [hunk 1]
```diff
@@ -82,4 +82,12 @@ export function scrollToExpense() {
   } else {
    window.scrollTo(0, 0);
   }
+}
+
+export function getUrlHash() {
+    return window.location.hash ? window.location.hash.substr(1).trim().toLowerCase().replace('-', ' ') : '';
```
Comment (xdamman) [inline frontend/src/lib/utils.js]: shouldn't we also remove the query params  with `hash.split('?')[0]`?

File: frontend/src/components/PaypalPreapproval.js [hunk 2]
```diff
@@ -2,43 +2,46 @@ import React, { Component, PropTypes } from 'react';
...
       </div>
     );
   }
 }
 
-PaypalReminder.propTypes = {
-  card: PropTypes.object,
+PaypalPreapproval.propTypes = {
+  i18n: PropTypes.object.isRequired,
   onClickConnect: PropTypes.bool.isRequired,
```
Comment (asood123) [inline frontend/src/components/PaypalPreapproval.js]: fixed.

File: frontend/src/components/PaypalPreapproval.js [hunk 3]
```diff
@@ -2,43 +2,46 @@ import React, { Component, PropTypes } from 'react';
...
+  card: PropTypes.object,
   onClickInProgress: PropTypes.bool,
-  i18n: PropTypes.object.isRequired,
+  loading: PropTypes.bool,
```
Comment (asood123) [inline frontend/src/components/PaypalPreapproval.js]: The component doesn't know when we fetch the paypalCard, so how does it distinguish between no paypal card and loading state on its own?

File: frontend/src/lib/utils.js (hunk 1 above)
Comment (asood123) [inline frontend/src/lib/utils.js]: yea, good idea

File: frontend/src/components/PaypalPreapproval.js [hunk 4]
```diff
@@ -2,43 +2,46 @@ import React, { Component, PropTypes } from 'react';
 
 import ConnectPaypalButton from './ConnectPaypalButton';
 
-class PaypalReminder extends Component {
+class PaypalPreapproval extends Component {
   render() {
     const {
       card,
       onClickConnect,
       onClickInProgress,
```
Comment (asood123) [inline frontend/src/components/PaypalPreapproval.js]: You mean onClick?

Comment (xdamman) [inline frontend/src/components/PaypalPreapproval.js]: onClick and onChange if needed, standard things.

**Pull request merged**